marketo/url.go

```diff
@@ -12,32 +12,31 @@ import (
 var restAPIPrefix string = "rest" //nolint:gochecknoglobals
 
 func (c *Connector) getURL(params common.ReadParams) (*urlbuilder.URL, error) {
-	// If NextPage is set, then we're reading the next page of results.
-	// The NextPage URL has all the necessary parameters.
-	if len(params.NextPage) > 0 {
-		return urlbuilder.New(params.NextPage.String())
-	}
-
 	link, err := c.getAPIURL(params.ObjectName)
 	if err != nil {
 		return nil, err
 	}
 
-	// This affects  a very few number of objects.
-	// Leads, Deleted Leads, Lead Changes,
-	if !params.Since.IsZero() {
-		time := params.Since.Format(time.RFC3339)
-		fmtTime := fmt.Sprintf("%v", time)
-		link.WithQueryParam("sinceDatetime", fmtTime)
+	// If NextPage is set, then we're reading the next page of results.
+	if len(params.NextPage) > 0 {
+		link.WithQueryParam("nextPageToken", params.NextPage.String())
+	}
+
+	// The only objects in Assets API supporting this are: Emails, Programs, SmartCampaigns,SmartLists
+	if !params.Since.IsZero() && c.Module == ModuleAssets.String() {
```

Comment:
Would be nice if we can do something like this to make it even more readable - 
```
if !params.Since.IsZero() {
  switch c.Module {
    case ModuleAssets.String()
    case ...
  }
}
```